### PR TITLE
Manage UpdateProductTagsCommand in separate command / handler

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Refacto UpdateProductOptionsCommand pr #19732 - separate command for updating tags, add missing behat assertion to assert that tags were not changed after invalid input, trim exploded strings in behat, make command input totally scalar. (previously it accepted array of objects, now - array of arrays), add constraint error code FAILED_UPDATE_TAGS
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? |  no
| Fixed ticket? | Fixes #{issue number here}.
| How to test?  | none

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19852)

### CODE REVIEW

File: src/Core/Domain/Product/Command/UpdateProductTagsCommand.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    /**
+     * @return LocalizedTags[]
+     */
+    public function getLocalizedTagsList(): array
+    {
+        return $this->localizedTagsList;
+    }
+
+    /**
+     * @param array $localizedTags
```
Comment (jolelievre) [inline src/Core/Domain/Product/Command/UpdateProductTagsCommand.php]: A description of the expected format would be nice

File: tests/Integration/Behaviour/Features/Scenario/Product/update_tags.feature [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    And product "product3" localized "tags" should be "en-US:mechanic,watch; fr-FR:"
+    When I update product "product3" tags with following values:
+      | tags       | fr-FR:montre |
+    And product "product3" localized "tags" should be "en-US:mechanic,watch; fr-FR:montre"
+
+  Scenario: Update product tags with invalid values
+    Given product "product3" localized "tags" should be "en-US:mechanic,watch; fr-FR:montre"
+    When I update product "product3" tags with following values:
+      | tags       | en-US:#<{ |
+    Then I should get error that product tag is invalid
```
Comment (jolelievre) [inline tests/Integration/Behaviour/Features/Scenario/Product/update_tags.feature:32]: Can you add the test we talked about to validate that the previous tags were not removed?

Comment (zuk3975) [reply tests/Integration/Behaviour/Features/Scenario/Product/update_tags.feature:32]: im pretty sure I've added it, but its gone :man_astronaut:

**Review by matks — APPROVED** [review]
Flawless

**Review by jolelievre — APPROVED** [review]
Thanks @zuk3975

**Pull request merged**